pilot/pkg/serviceregistry/kube/controller/ambientindex.go

```diff
@@ -190,15 +189,32 @@ func (a *AmbientIndexImpl) All() []*model.AddressInfo {
 	a.mu.RLock()
 	defer a.mu.RUnlock()
 	res := make([]*model.AddressInfo, 0, len(a.byUID)+len(a.serviceByNamespacedHostname))
+
 	for _, wl := range a.byUID {
+		netAddrs := networkAddressFromWorkload(wl)
```

Comment:
Do we need to create a parsed copy of the addresses? Can we just use the raw wl.Addresses. Seems expensive to do this for all workloads on the read side.

Reply:
It's because the keys of the lookup maps `byPod` and `byWorkloadEntry` are the parsed addresses.